This quick'n'dirty patch enables relatively painless SpiderMonkey installation.
See doc/ecmascript.txt for details.

diff -ru js-1.5-rc6a/src/config.mk js-1.5-rc6a+elinks/src/config.mk
--- js-1.5-rc6a/src/config.mk	Sat Nov 15 01:10:55 2003
+++ js-1.5-rc6a+elinks/src/config.mk	Wed Sep 22 16:32:12 2004
@@ -37,10 +37,15 @@
 # 
 # ***** END LICENSE BLOCK *****
 
+BUILD_OPT=1
+ifndef PREFIX
+PREFIX = /opt/spidermonkey
+endif
+
 ifdef JS_DIST
 DIST = $(JS_DIST)
 else
-DIST = $(DEPTH)/../../dist/$(OBJDIR)
+DIST = dist/
 endif
 
 # Set os+release dependent make variables
diff -ru js-1.5-rc6a/src/jsconfig.mk js-1.5-rc6a+elinks/src/jsconfig.mk
--- js-1.5-rc6a/src/jsconfig.mk	Sat Nov 15 01:10:56 2003
+++ js-1.5-rc6a+elinks/src/jsconfig.mk	Wed Sep 22 16:34:28 2004
@@ -104,8 +104,8 @@
 	echo $(NSPR_VERSION) > $(NSPR_VERSIONFILE)
 endif
 
-SHIP_DIST  = $(MOZ_DEPTH)/dist/$(OBJDIR)
-SHIP_DIR   = $(SHIP_DIST)/SHIP
+SHIP_DIST  = $(DIST)
+SHIP_DIR   = $(PREFIX)
 
 SHIP_LIBS      = libjs.$(SO_SUFFIX) libjs.a
 ifdef JS_LIVECONNECT
@@ -117,7 +117,7 @@
     SHIP_LIBS += jsj.dll jsj.lib
   endif
 endif
-SHIP_LIBS     += $(LCJAR)
+#SHIP_LIBS     += $(LCJAR)
 SHIP_LIBS     := $(addprefix $(SHIP_DIST)/lib/, $(SHIP_LIBS))
 
 SHIP_INCS      = js*.h prmjtime.h resource.h *.msg *.tbl
@@ -144,6 +144,14 @@
   JSREFJAR = jsref_dbg.jar
 endif
 endif
+
+install:
+	mkdir -p $(SHIP_DIR)/$(LIBDIR)
+	mkdir -p $(SHIP_DIR)/include
+	mkdir -p $(SHIP_DIR)/bin
+	cp $(SHIP_LIBS) $(SHIP_DIR)/$(LIBDIR)
+	cp $(SHIP_INCS) $(SHIP_DIR)/include
+	cp $(SHIP_BINS) $(SHIP_DIR)/bin
 
 ship:
 	mkdir -p $(SHIP_DIR)/$(LIBDIR)
